src/unix/lwt_io.mli

```diff
@@ -404,6 +404,14 @@ val with_file :
       file and passes the channel to [f]. It is ensured that the
       channel is closed when [f ch] terminates (even if it fails). *)
 
+val open_temp_file :
+  ?buffer:Lwt_bytes.t ->
+  ?perm:Unix.file_perm -> unit -> output_channel Lwt.t
```

Comment:
At least this function should evaluate to a `(string, output_channel) Lwt.t`. With this interface, I believe there is no way for the caller to delete the file afterwards. I believe this issue is why [Filename.open_temp_file](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Filename.html#VALopen_temp_file) was added. `with_temp_file` should also pass the filename to its callback.